envi/archs/arm/const.py

```diff
@@ -3,6 +3,63 @@
 MODE_JAZELLE    = 2
 MODE_THUMBEE    = 3
 
+'''
+Support for different ARM Instruction set versions
+Note that the bit values COULD change as more are added so always use the name when referencing versions
+'''
+# name          bitmask                    decimal         hex
+REV_ARMv4   =   0b0000000000000000000001 #        1        0x1
+REV_ARMv4T  =   0b0000000000000000000010 #        2        0x2
+REV_ARMv5   =   0b0000000000000000000100 #        4        0x4
+REV_ARMv5T  =   0b0000000000000000001000 #        8        0x8
+REV_ARMv5E  =   0b0000000000000000010000 #       16        0x10
+REV_ARMv5J  =   0b0000000000000000100000 #       32        0x20
+REV_ARMv5TE =   0b0000000000000001000000 #       64        0x40
+REV_ARMv6   =   0b0000000000000010000000 #      128        0x80
+REV_ARMv6T2 =   0b0000000000000100000000 #      256        0x100
+REV_ARMv6M  =   0b0000000000001000000000 #      512        0x200
+REV_ARMv7A  =   0b0000000000010000000000 #     1024        0x400
+REV_ARMv7R  =   0b0000000000100000000000 #     2048        0x800
+REV_ARMv7M  =   0b0000000001000000000000 #     4096        0x1000
+REV_ARMv7EM =   0b0000000010000000000000 #     8192        0x2000
+REV_ARMv8A  =   0b0000000100000000000000 #    16384        0x4000
+REV_ARMv8R  =   0b0000001000000000000000 #    32768        0x8000
+REV_ARMv8M  =   0b0000010000000000000000 #    65536        0x10000
+REVS_ARMv4  = (REV_ARMv4 | REV_ARMv4T)
+REVS_ARMv5  = (REV_ARMv5 | REV_ARMv5T | REV_ARMv5E | REV_ARMv5J | REV_ARMv5TE)
+REVS_ARMv6  = (REV_ARMv6 | REV_ARMv6T2 | REV_ARMv6M)
+REVS_ARMv7  = (REV_ARMv7A | REV_ARMv7R | REV_ARMv7M | REV_ARMv7EM) 
+REVS_ARMv8  = (REV_ARMv8A | REV_ARMv8R | REV_ARMv8M)
+REV_ALL = (REVS_ARMv4 | REVS_ARMv5 | REVS_ARMv6 | REVS_ARMv7 | REVS_ARMv8)
+
+#Not sure, did from memory , needs to be confirmed
+REVT_THUMB16 = (REVS_ARMv5 | REVS_ARMv6)
+REVT_THUMB2  = (REVS_ARMv7 | REVS_ARMv8)
+REVT_THUMBEE = (REVS_ARMv7 | REVS_ARMv8)
+
+#In progress - Draft version
+ARCH_REVS = {
```

Comment:
No problem. I will make it so.. 
Will make other changes as discussed and will roll this into the unittest PR and we can close out the const PR. 




yes, comprehensions are fast.  this is done once at startup.  it's not done repeatedly.  for such code, we weigh the value of readability against speed.  in this case, i think the readability is better.  we can then explain what's happening in a comment.  for code that is executed in a tight loop repeatedly, the speed is more important.